src/boot/efi/meson.build

```diff
@@ -33,8 +33,16 @@ stub_sources = '''
 '''.split()
 
 if conf.get('ENABLE_EFI') == 1 and get_option('gnu-efi') != 'false'
-        efi_cc = get_option('efi-cc')
-        efi_ld = get_option('efi-ld')
+        if get_option('efi-cc') == '$cc'
+                efi_cc = ' '.join(cc.cmd_array())
+        else
+                efi_cc = get_option('efi-cc')
+        endif
+        if get_option('efi-ld') == '$ld'
+                efi_ld = find_program('ld', required: true)
+        else
+                efi_ld = get_option('efi-ld')
+        endif
```

Comment:
Maybe
```meson
efi_ld = get_option('efi-ld')
if efi_ld == ''
     efi_ld = find_program('ld', required: true)
endif
```
?

Reply:
@helmutg since this is your patch, it would be great if you can chime in here